jax/_src/nn/functions.py

```diff
@@ -317,20 +329,22 @@ def gelu(x: Array, approximate: bool = True) -> Array:
     x : input array
     approximate: whether to use the approximate or exact formulation.
   """
+  numpy_util.check_arraylike("gelu", x)
 
   # Promote to nearest float-like dtype.
-  x = x.astype(dtypes.to_inexact_dtype(x.dtype))
+  x_arr = jnp.asarray(x)
+  x_arr = x_arr.astype(dtypes.to_inexact_dtype(x_arr.dtype))
```

Comment:
You could replace these three lines with `x_arr = numpy_util.promote_args_inexact("gelu", x)`

Reply:
Nice, done!